src/main.ts

```diff
@@ -302,110 +324,164 @@ async function getTsServerRepoResult(
             }
         }
 
-        const owner = repo.owner ? `${mdEscape(repo.owner)}/` : "";
-        const url = repo.url ? `(${repo.url})` : "";
+        const tsServerResult = {
+            oldServerFailed,
+            oldSpawnResult,
+            newServerFailed,
+            newSpawnResult,
+            replayScriptPath,
+            installCommands,
+        };
 
-        let summary = `## [${owner}${mdEscape(repo.name)}]${url}\n`;
+        if (oldServerFailed && !newServerFailed) {
+            return { status: "Detected interesting changes", tsServerResult }
+        }
+        if (!newServerFailed) {
+            return { status: "Detected no interesting changes" };
+        }
 
-        if (oldServerFailed) {
-            const oldServerError = oldSpawnResult?.stdout
-                ? prettyPrintServerHarnessOutput(oldSpawnResult.stdout, /*filter*/ true)
-                : `Timed out after ${executionTimeout} ms`;
-            summary += `
+        return { status: "Detected interesting changes", tsServerResult, replayScriptPath, rawErrorPath };
+    }
+    catch (err) {
+        reportError(err, `Error running tsserver on ${repo.url ?? repo.name}`);
+        return { status: "Unknown failure" };
+    }
+    finally {
+        console.log(`Done ${repo.url ?? repo.name}`);
+        logStepTime(diagnosticOutput, repo, "language service", lsStart);
+    }
+}
+
+function groupOldErrors(summaries: Summary[]): Map<string, Summary[]> {
+    const group = new Map<string, Summary[]>();
+    for (const summary of summaries) {
+        if (summary.tsServerResult.oldServerFailed && !summary.tsServerResult.newServerFailed) {
+            const error = parseServerHarnessOutput(summary.tsServerResult.oldSpawnResult!.stdout);
```

Comment:
This change from a `!.` seems suspicious to me. The old code used to use a `?.`, implying that there were cases where `oldServerFailed` might be true, but `oldSpawnResult` might not be present.

Reply:
Also, does this assume that the old result and the new result is equal? Does it need to compare the contents of `stdout` between `oldSpawnResult` and `newSpawnResult`?

Reply:
> This change from a `!.` seems suspicious to me. The old code used to use a `?.`, implying that there were cases where `oldServerFailed` might be true, but `oldSpawnResult` might not be present.

Good catch, `oldSpawnResult` can be `undefined` if the server failed due to a timeout. Fixed.

> Also, does this assume that the old result and the new result is equal? Does it need to compare the contents of `stdout` between `oldSpawnResult` and `newSpawnResult`?

I'm not sure if I'm understanding this correctly, this is only to report that the old server used to show an error and is no longer the case. Whether is intended or not is discretionary of the dev.